src/AutoRest.CSharp/MgmtTest/Generation/MgmtExtensionTestWriter.cs

```diff
@@ -0,0 +1,140 @@
+                _writer.Line($"System.Environment.SetEnvironmentVariable(\"RESOURCE_MANAGER_URL\", $\"https://localhost:8443\");");
+            }
+        }
+
+        protected void WriteMethodTest(MgmtClientOperation clientOperation, bool async, bool isLroOperation)
+        {
+            Debug.Assert(clientOperation != null);
+            var methodName = clientOperation.Name;
+            int exampleIdx = 0;
+            foreach ((var branch, var operation) in GetSortedOperationMappings(clientOperation))
+            {
+                var exampleGroup = MgmtBaseTestWriter.FindExampleGroup(operation);
+                if (exampleGroup is null || exampleGroup.Examples.Count() == 0)
+                    return;
+
+                foreach (var exampleModel in exampleGroup?.Examples ?? Enumerable.Empty<ExampleModel>())
+                {
+                    WriteTestDecorator();
+                    var testCaseSuffix = exampleIdx > 0 ? (exampleIdx + 1).ToString() : String.Empty;
+                    _writer.Append($"public {GetAsyncKeyword(async)} {MgmtBaseTestWriter.GetTaskOrVoid(async)} {methodName}{testCaseSuffix}()");
+                    using (_writer.Scope())
+                    {
+                        _writer.LineRaw($"// Example: {exampleModel.Name}");
+                        WriteOperationInvocation(clientOperation, operation, exampleModel, async, isLroOperation);
+                    }
+                    _writer.Line();
+                    exampleIdx++;
+                }
+            }
+        }
+
+        public void WriteOperationInvocation(MgmtClientOperation clientOperation, MgmtRestOperation restOperation, ExampleModel exampleModel, bool async, bool isLroOperation)
+        {
+            var testMethodName = CreateMethodName(clientOperation.Name, async);
+            var resourceIdentifierParams = ComposeResourceIdentifierParams(restOperation.RequestPath, exampleModel);
+            var subscriptionVariableName = new CodeWriterDeclaration(extensions.Type.Name.FirstCharToLowerCase());
+            var subscriptionRequestPath = GetSubscriptionRequestPath(restOperation.RequestPath.SerializedPath, exampleModel);
+            _writer.Line($@"var {subscriptionVariableName:D} = GetArmClient().GetSubscription(new {typeof(Azure.Core.ResourceIdentifier)}({FormatResourceId(subscriptionRequestPath).RefScenariDefinedVariables(scenarioVariables)}));");
+            List<KeyValuePair<string, FormattableString>> parameterValues = WriteOperationParameters(clientOperation.MethodParameters.ToArray().AsSpan().Slice(1).ToArray(), exampleModel);
+
+            _writer.Line();
+            WriteMethodTestInvocation(async, clientOperation, isLroOperation, $"{subscriptionVariableName}.{testMethodName}", parameterValues.Select(pv => pv.Value));
+        }
+
+        public string GetSubscriptionRequestPath(string requestPath, ExampleModel exampleModel)
```

Comment:
This method deals with quite a few low level things - but I believe we could use the `RequestPath` class instead of passing in raw string to simplify it.